Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalInfoSamplerTests.swift

```diff
@@ -67,6 +67,32 @@ class VitalInfoSamplerTests: XCTestCase {
         }
     }
 
+    func testItDoesSamplePeriodicallyWithLowFrequency() {
+        let mockCPUReader = SamplingBasedVitalReaderMock()
+        let mockMemoryReader = SamplingBasedVitalReaderMock()
+
+        let sampler = VitalInfoSampler(
+            cpuReader: mockCPUReader,
+            memoryReader: mockMemoryReader,
+            refreshRateReader: ContinuousVitalReaderMock(),
+            frequency: 0.5
+        )
+
+        mockCPUReader.vitalData = 123.0
+        mockMemoryReader.vitalData = 321.0
+
+        let samplingExpectation = expectation(description: "sampling expectation")
+        DispatchQueue.global().asyncAfter(deadline: .now() + 0.6) {
+            samplingExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 1.0) { _ in
+            // 2 samples, initial sample and one periodic sample
+            XCTAssertEqual(sampler.cpu.sampleCount, 2)
+            XCTAssertEqual(sampler.memory.sampleCount, 2)
+        }
```

Comment:
I have concerns about reliability of this test - it looks flaky. Does it pass if ran with 100x repetitions locally? That could be some indicator of flakiness.

What concerns me is the configuration, notably:
* `VitalInfoSampler(..., frequency: 0.5, ...)`
* `asyncAfter(deadline: .now() + 0.6) { samplingExpectation.fulfill() }`

It seems that this test is built around assumption that both initial and 1st sample will happen within `0.6s` - any lag longer than `0.1s` could make it fail, no?

Reply:
It's the same as it was before this change. The value was just 1 instead of 2. The initial sample wasn't getting picked up because it was fired before the mock values were set:
```
mockCPUReader.vitalData = 123.0
mockMemoryReader.vitalData = 321.0
```

Git isn't very clear, but this is the old test (fixed), the new test is the one above it.

Reply:
I run it 100 times and it seems to be good:
```
Test Suite 'VitalInfoSamplerTests' passed at 2023-03-01 17:57:01.387.
	 Executed 400 tests, with 0 failures (0 unexpected) in 173.177 (173.642) seconds
Test Suite 'DatadogTests.xctest' passed at 2023-03-01 17:57:01.388.
	 Executed 400 tests, with 0 failures (0 unexpected) in 173.177 (173.643) seconds
Test Suite 'Selected tests' passed at 2023-03-01 17:57:01.389.
	 Executed 400 tests, with 0 failures (0 unexpected) in 173.177 (173.646) seconds
```

Proceeding with merge :)